pkg/api/types.go

```diff
@@ -140,7 +140,9 @@ type ObjectMeta struct {
 	Annotations map[string]string `json:"annotations,omitempty"`
 
 	// List of objects depended by this object. If ALL objects in the list have
-	// been deleted, this object will be garbage collected.
+	// been deleted, this object will be garbage collected. If this object is managed
+	// by a controller and OwnerReference pointing to this controller has a controller filed
+	// set to true. There can be only one managing controller.
```

Comment:
I suggested the "first entry" constraint, but maybe that's not such a good idea. I'm not sure the user could guarantee that if using `kubectl apply`, which does a mechanical list merge.